rust/tvm/src/ir/module.rs

```diff
@@ -25,16 +29,20 @@ use crate::runtime::{external, Object, ObjectRef};
 
 use super::expr::GlobalVar;
 use super::function::BaseFunc;
-
-use std::io::Result as IOResult;
-use std::path::Path;
-
-use tvm_macros::Object;
+use super::source_map::SourceMap;
 
 // TODO(@jroesch): define type
 type TypeData = ObjectRef;
 type GlobalTypeVar = ObjectRef;
 
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("{0}")]
```

Comment:
Do we also want to use `#[source]` for the errors?

Reply:
This kind of stuff would be good follow up work imo.